### PR TITLE
chore(karpenter-core): go get sigs.k8s.io/karpenter@v0.32.2-0.20231205044115-1cf3df4ba3cc (5/6)

### PR DESCRIPTION
**Pull request by charliedmcb**
Fixes # 

**Description**

This core bump is based off of the following `aws` PR:
- https://github.com/aws/karpenter-provider-aws/commit/dafe148a327a8df619955bef0a8150edd27805dd#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6

**How was this change tested?**

*

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates
- [ ] Yes, issue opened: # 
- [ ] No

**Release Note**

```release-note

```

### CODE REVIEW

File: pkg/apis/v1alpha2/labels.go [hunk 1]
```diff
@@ -119,8 +118,6 @@ var (
 		'c': LabelSKUConfidential,
 		'i': LabelSKUIsolatedSize,
 	}
-
-	NodeClaimLinkedAnnotationKey = v1alpha5.MachineLinkedAnnotationKey // still using the one from v1alpha5
```
Comment (charliedmcb) [inline pkg/apis/v1alpha2/labels.go:123]: I did see anywhere we reference this, but is it needed elsewhere that we should keep this around for some reason?

Would need to hardcode the value now though.

Comment (Bryce-Soghigian) [reply pkg/apis/v1alpha2/labels.go:123]: No, this value is useless now without Link Controller @rakechill  can correct me if im wrong

File: .github/workflows/approval-comment.yaml [hunk 2]
```diff
@@ -26,7 +26,7 @@ jobs:
           mkdir -p /tmp/artifacts
           { echo ${{ github.event.pull_request.number }}; echo ${{ github.event.review.commit_id }}; } >> /tmp/artifacts/metadata.txt
           cat /tmp/artifacts/metadata.txt
-      - uses: actions/upload-artifact@5d5d22a31266ced268874388b861e4b58bb5c2f3 # v4.3.1
+      - uses: actions/upload-artifact@26f96dfa697d77e81fd5907df203aa23a56210a8 # v4.3.0
```
Comment (tallaxes) [inline .github/workflows/approval-comment.yaml:29]: We don't want to downgrade the version here

File: .github/workflows/codeql-analysis.yml [hunk 3]
```diff
@@ -33,8 +33,8 @@ jobs:
       - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
       - uses: ./.github/actions/install-deps
       - run: make vulncheck
-      - uses: github/codeql-action/init@e8893c57a1f3a2b659b6b55564fdfdbbd2982911 # v3.24.0
+      - uses: github/codeql-action/init@b7bf0a3ed3ecfa44160715d7c442788f65f0f923 # v3.23.2
```
Comment (tallaxes) [inline .github/workflows/codeql-analysis.yml:36]: Don't want to downgrade the version here (and elsewhere)

File: charts/karpenter/templates/webhooks-core.yaml
Comment (tallaxes) [inline charts/karpenter/templates/webhooks-core.yaml:1]: There is more v1alpha5 stuff to cleanup in Helm charts, is the plan to handle it separately?

Comment (Bryce-Soghigian) [reply charts/karpenter/templates/webhooks-core.yaml:1]: https://github.com/Azure/karpenter-provider-azure/pull/152  is my attempt to clean some of that up for him

Comment (Bryce-Soghigian) [reply charts/karpenter/templates/webhooks-core.yaml:1]: Another question is do we want to completely remove webhooks from chart or allow them for self hosted and leave it as a perk for using self hosted?

**Review by tallaxes — APPROVED** [review]
Left a couple of comments (re plans for further Helm chart cleanup; also not sure if action version issue will be resolved in the final PR against main), otherwise LGTM

File: charts/karpenter/templates/webhooks-core.yaml
Comment (tallaxes) [reply charts/karpenter/templates/webhooks-core.yaml:1]: I would not remove them (yet), but would disable them by default

Comment (charliedmcb) [reply charts/karpenter/templates/webhooks-core.yaml:1]: I do also remove some in this PR too:
- https://github.com/Azure/karpenter-provider-azure/pull/151/files#diff-6dbf6368d06589ba259f1c1202ddf04eb43ce306f755a039d654221d1093b544

However, fair it'd be best to probably migrate the removals to this PR

File: .github/workflows/codeql-analysis.yml (hunk 3 above)
Comment (charliedmcb) [reply .github/workflows/codeql-analysis.yml]: Fixed

File: .github/workflows/approval-comment.yaml (hunk 2 above)
Comment (charliedmcb) [reply .github/workflows/approval-comment.yaml]: Fixed, along with the others in `.github/workflows`

File: charts/karpenter/templates/webhooks-core.yaml
Comment (charliedmcb) [reply charts/karpenter/templates/webhooks-core.yaml:1]: Removed remaining v1alpha5. However, didn't go around purging anything referencing Machines/Provisioners.

Happy to help with your follow up PR @Bryce-Soghigian

Comment (charliedmcb) [reply charts/karpenter/templates/webhooks-core.yaml:1]: Looking if there's any final opinions on the purge of `v1alpha5`

Comment (Bryce-Soghigian) [reply charts/karpenter/templates/webhooks-core.yaml:1]: merge

**Pull request merged**